src/api/routes/sports.ts

```diff
@@ -20,11 +20,30 @@ router.get("/", (req: Request, res: Response, next: NextFunction) => {
 router.get("/:category", (req: Request, res: Response, next: NextFunction) => {
     try {
         const category: any = req.params.category;
-        const result = data.data.filter((sport: any) => sport.category === category);
-        if (result.length)
-            res.status(200).json(result);
-        else
-            res.status(404).json({ message: "Invalid category" });
+        const player: any = req.query.players;
+        const result = data.data.filter((sport: any) => sport.category === category)[0];
+        if (result && Object.keys(req.query).length) {
+            if (!player)
+                res.status(404).json({ message: "Invalid query" });
+            else {
+                const ans: any = [];
+                const length = result.players.length;
+                for (let i: any = 0; i < length; i++) {
```

Comment:
Remove the unecesaary variable `length`. It can be reduced to one line as 
```js
for (let i: any = 0; i < result.players.length; i++)
```

Reply:
Done!